robot/basestation/static/js/elements/feed/navbar.js

```diff
@@ -86,151 +111,71 @@ $(document).ready(() => {
-      requestMuxChannel('#mux-2', function (msgs) {
+      requestMuxChannel('#' + muxId, function (msgs) {
         printErrToConsole(msgs)
 
-        if (msgs[0] == true && window.location.pathname == '/science') {
-          console.log('Activating Science Listener Node')
-
-          let serialType = getCookie('serialType')
-
-          if (serialType == '') {
-            appendToConsole('Serial type not yet defined!')
-            return
-          }
-
-          // automating opening listener and sending MCU ping in UART mode
-          if (serialType == 'uart') {
-            requestTask(
-              SCIENCE_LISTENER_TASK,
-              STATUS_START,
-              function (msgs) {
-                if (msgs[0]) {
-                  $('#science-listener-btn')[0].checked = true
-                  // try pinging MCU
-                  wait(1000)
-                  sendRequest('Science', 'ping', printErrToConsole)
-                } else {
-                  $('#science-listener-btn')[0].checked = false
-                }
-              },
-              serialType
-            )
+        if (muxId != 'mux-3') {
+          if (msgs[0]) {
+            rosLog(ROSINFO, 'Activating ' + deviceName + ' Listener Node')
+
+            let serialType = getCookie('serialType')
+
+            if (serialType == '') {
+              rosLog(ROSINFO, 'Serial type not yet defined!')
+              return
+            }
+
+            // automating opening listener and sending MCU ping in UART mode
+            if (serialType == 'uart') {
+              requestTask(
+                ROVER_LISTENER_TASK,
+                STATUS_START,
+                function (msgs) {
+                  if (msgs[0]) {
+                    $('#toggle-rover-listener-btn')[0].checked = true
+                    // try pinging MCU
+                    wait(1000)
+                    sendRequest(deviceName, 'ping', printErrToConsole)
+                  } else {
+                    $('#toggle-rover-listener-btn')[0].checked = false
+                  }
+                },
+                serialType
+              )
+            }
           }
         }
       })
     }
-  })
+  }
 
-  $('#mux-3').mouseup(function () {
-    // PDS
-    if (isListenerOpen() && getCookie('serialType') == 'uart') {
-      appendToConsole('Don\'t change the mux channel while a listener is open!')
-    } else {
-      requestMuxChannel('#mux-3', function (msgs) {
-        printErrToConsole(msgs)
-      })
-    }
-  })
+  // set serialtype
+  $('#uart').mouseup(setSerialUart)
+  $('#usb').mouseup(setSerialUsb)
 
-  $('#uart').mouseup(function () {
+  function setSerialUart () {
     $('#serial-type').text('uart')
     setCookie('serialType', 'uart', 3)
-    appendToConsole('setting cookie to uart')
-  })
+    rosLog(ROSINFO, 'setting cookie to uart')
+  }
 
-  $('#usb').mouseup(function () {
+  function setSerialUsb () {
     $('#serial-type').text('usb')
     setCookie('serialType', 'usb', 3)
```

Comment:
```suggestion
    setCookie('serialType', 'usb', 356)
```

Reply:
Suggesting this change as I found it a bit of a hassle during comp to have to reset these values rather than keeping the previous ones active, since the missions are spread out over a week usually.

Reply:
fixed